apps/omg_watcher/lib/exit_processor/in_flight_exit_info.ex

```diff
@@ -0,0 +1,222 @@
+# Copyright 2018 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Watcher.ExitProcessor.InFlightExitInfo do
+  @moduledoc """
+  Represents the bulk of information about a tracked in-flight exit.
+
+  Internal stuff of `OMG.Watcher.ExitProcessor`
+  """
+
+  alias OMG.API.State.Transaction
+
+  @output_index_range 0..7
+
+  @block_offset 1_000_000_000
+  @tx_offset 10_000
+
+  defstruct [
+    :tx,
+    :tx_pos,
+    :timestamp,
+    :contract_id,
+    :oldest_competitor,
+    # piggybacking
+    exit_map:
+      @output_index_range
+      |> Enum.map(&{&1, %{is_piggybacked: false, is_finalized: false}})
+      |> Map.new(),
+    is_canonical: true,
+    is_active: true
+  ]
+
+  @type tx_position() :: {pos_integer(), non_neg_integer()}
+  @type ife_contract_id() :: <<_::192>>
+
+  @type t :: %__MODULE__{
+          tx: Transaction.Signed.t(),
+          tx_pos: tx_position(),
+          timestamp: non_neg_integer(),
+          contract_id: ife_contract_id(),
+          oldest_competitor: {non_neg_integer(), non_neg_integer()},
```

Comment:
What are those `non_neg_integers()`? Data structures are central to understanding - lets document them thoroughly.

Reply:
Also, incomplete type - `nil` is also used as value.